src/cider/nrepl/middleware/spec.clj

```diff
@@ -0,0 +1,128 @@
+(ns cider.nrepl.middleware.spec
+  (:require [cider.nrepl.middleware.util.error-handling :refer [with-safe-transport]]
+            [cider.nrepl.middleware.util.spec :as spec-utils]
+            [clojure.tools.nrepl
+             [middleware :refer [set-descriptor!]]
+             [misc :refer [response-for]]]
+            [cider.nrepl.middleware.util.cljs :as cljs]
+            [clojure.string :as str]
+            [clojure.walk :as walk]
+            [clojure.pprint :as pp]))
+
+(defn str-non-colls
+  "Given a form, convert all non collection childs to str"
```

Comment:
All docstring should end with a `.`.

I'd also use full words like string instead of `str` - that's typically  reserved for a parameter name.